qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -133,25 +135,54 @@ def create_reqd_users(self):
                                        check_status=False)
 
     def write_local_config(self):
-        from os.path import join
-        from textwrap import dedent
+        if isinstance(self.mount_a, KernelMount):
+            conf_contents = self._gen_conf_for_kernel_mnt()
+        elif isinstance(self.mount_a, (FuseMount, LocalFuseMount)):
+            conf_contents = self._gen_conf_for_fuse_mnt()
 
+        self.mount_a.client_remote.write_file(join(self.repo_path,
+                                                   'local.config'),
+                                              conf_contents, sudo=True)
+
+    # generate local.config for kernel cephfs mounts
+    def _gen_conf_for_kernel_mnt(self):
         mon_sock = self.fs.mon_manager.get_msgrv1_mon_socks()[0]
-        self.test_dev = mon_sock + ':/' + self.test_dirname
-        self.scratch_dev = mon_sock + ':/' + self.scratch_dirname
+        test_dev = mon_sock + ':/' + self.test_dirname
+        scratch_dev = mon_sock + ':/' + self.scratch_dirname
 
-        xfstests_config_contents = dedent('''\
+        return dedent(f'''\
             export FSTYP=ceph
-            export TEST_DEV={}
-            export TEST_DIR={}
-            #export SCRATCH_DEV={}
-            #export SCRATCH_MNT={}
-            export TEST_FS_MOUNT_OPTS="-o name=admin,secret={}"
-            ''').format(self.test_dev, self.test_dirs_mount_path, self.scratch_dev,
-                        self.scratch_dirs_mount_path, self.get_admin_key())
-
-        self.mount_a.client_remote.write_file(join(self.repo_path, 'local.config'),
-                                              xfstests_config_contents, sudo=True)
+            export TEST_DEV={test_dev}
+            export TEST_DIR={self.test_dirs_mount_path}
+            export TEST_FS_MOUNT_OPTS="-o name=admin,secret={self.get_admin_key()}"
+            #export SCRATCH_DEV={scratch_dev}
+            #export SCRATCH_MNT={self.scratch_dirs_mount_path}''')
+
+    # generate local.config for FUSE cephfs mounts
+    def _gen_conf_for_fuse_mnt(self):
+        mon_sock = self.fs.mon_manager.get_msgrv1_mon_socks()[0]
+        test_dev = 'ceph-fuse'
```

Comment:
What the `test_dev` will be used for ceph-fuse case ?  Why set it to `ceph-fuse` ?

Reply:
This is still a little confusing for me.

As my understanding the `test_dev` should be raw disks something like `/dev/sda`, and it should be `mon_sock + ':/' + self.test_dirname` for ceph.

Could explain more about it ? Since the `test_dev` will be used in many test cases, I'm wondering whether will it cause issues ?

Reply:
From the IRC, since this is just a placeholder, IMO you should set it to `mon_sock + ':/' + self.test_dirname` anyway to avoid introducing confusion. This is what it should be.

Reply:
Won't that give a false impression that the device is being used?

Reply:
I think I missed this.

BTW, what will happen if we just not set the `TEST_DEV` and `SCRATCH_DEV` for fuse?

Reply:
`TEST_DEV` is mandatory as per xfstests-dev -

```
2022-12-14 19:00:35,019.019 DEBUG:__main__:> sudo ./check generic/099
Warning: need to define parameters for host xxxxx
       or set variables:
        TEST_DEV
```

`generic/099` works fine even if `SCRATCH_DEV` is left unset.

Reply:
Okay.

Reply:
@rishabh-d-dave I think @lxbsz is correct. setting test_dev to ceph-fuse is confusing.